en/05_Contributing/01_Code.md

```diff
@@ -6,245 +6,283 @@ icon: code
- - entities marked as `@internal`
- - yml configuration file default values
- - HTML, CSS, JavaScript, TypeScript, SQL and anything else that is not PHP
+#### For changes to public API or new/enhanced features
 
-Other entities might be considered to be included or excluded from the public APIs on case-by-case basis.
+If you are introducing new APIs, introducing new features, or enhancing an existing feature, you should generally use the default branch of the repository where you want to contribute to. That would usually target the next minor release of the module.
 
-Any updates to third party dependencies in composer.json should aim to target the default branch for a minor release if possible. Targeting a patch release branch is acceptable if updating dependencies is required to fix a bug.
+#### For bug fixes that don't introduce new API
 
-API from third party dependencies may indirectly be incorporated into Silverstripe CMS's API if:
-- they are defined as a parameter type for a supported method
-- they are defined as a return type for a supported method
-- they are extended by a Silverstripe CMS class.
+If you are fixing a bug that doesn't require API changes, use the highest patch release branch available for the lowest supported major release line the bug applies to. You can see the currently supported release lines for Silverstripe CMS on [the roadmap](https://www.silverstripe.org/software/roadmap/). You can find which major release lines of core and supported modules apply to that version by checking the relevant [/project_governance/supported_modules/](supported modules) page.
 
-When defining a return type or a parameter type, it is preferable to use a more generic interface rather than a specific class. Third party dependencies that are used for internal purposes and are not explicitly exposed via the Silverstripe CMS public API are not covered by SemVer and maybe substituted without notice.
+For example, if your bug fix is applicable for Silverstripe CMS 4, and is for the `silverstripe/admin` module, you would target the `1.13` branch.
 
-### The Pull Request Process
+#### For API breaking changes
 
-Once your pull request is issued, it's not the end of the road. A [core committer](/project_governance/core_committers/) will most likely have some questions for you and may ask you to make some changes depending on discussions you have.
-If you've been naughty and not adhered to the [coding conventions](coding_conventions), 
-expect a few requests to make changes so your code is in-line.
+Do not make a pull request that includes a breaking change, including changing public API (described below), unless there is a major release branch ready to merge into.
+e.g. if the latest stable release is `5.2.7`, the major release branch would be `6`.
 
-If your change is particularly significant, it may be referred to the [forum](https://forum.silverstripe.org) for further community discussion.
+#### Definition of public API
```

Comment:
A future PR will move this definition into a more fitting page - but for now, please review it as though this was the best place for it.